addons/sale_coupon/tests/test_program_numbers.py

```diff
@@ -1179,3 +1179,27 @@ def test_program_maximum_use_number_with_other_promo(self):
         order2.recompute_coupon_lines()
         self.assertEqual(order2.amount_total, 22.5, "10% discount should be applied")
         self.assertEqual(len(order2.order_line.ids), 2, "discount should be applied")
+
+    def test_program_maximum_use_number_last_order(self):
+        #reuse p1 with a different promo code and a maximum_use_number of 1
+        self.p1.promo_code = 'promo1'
```

Comment:
Good job reusing the already existing programs and products from the `setUp()` :+1: